crates/astria-conductor/src/executor.rs

```diff
@@ -156,12 +162,28 @@ impl<C: ExecutionClientV1Alpha1> Executor<C> {
                         );
                         continue;
                     };
-                    if let Err(e) = self.execute_block(block_subset).await {
-                        error!(
-                            height = height,
-                            error = ?e,
-                            "failed to execute block"
-                        );
+                    match self.execute_block(block_subset.clone()).await {
+                        Ok(Some(execution_block_hash)) => {
+                            // after execution, set SOFT as this block. FIRM stays the same.
+                            let timestamp =
+                                convert_tendermint_to_prost_timestamp(block_subset.header.time)?;
+                            let firm = self.commitment_state.firm.clone().unwrap();
+                            let soft = Block {
+                                number: height as u32,
+                                hash: execution_block_hash.clone(),
+                                parent_block_hash: firm.hash.clone(),
```

Comment:
I believe this should be `parent_block_hash: self.commitment_state.soft.unwrap().hash.clone()` or something like that. This blocks parent is the previous executed block, not necessarily the `firm` block.